test/e2e_test/endtoend_test.go

```diff
@@ -0,0 +1,101 @@
+package e2e_test
+
+import (
+	"context"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/sustglobal/gost/component"
+	"github.com/sustglobal/gost/event"
+	"github.com/sustglobal/gost/examples/sample_component"
+	"github.com/sustglobal/gost/impl/gcp"
+
+	"cloud.google.com/go/pubsub"
+	"google.golang.org/api/option"
+)
+
+type ServiceConfig struct {
+	CustomValue string `env:"CUSTOM_VALUE"`
+}
+
+type PubSubConfig struct {
+	GCPProjectID        string
+	GCPPubSubTopic      string
+	GCPSubscriptionName string
+	context             context.Context
+	options             []option.ClientOption
+}
+
+func TestE2E(t *testing.T) {
+	os.Setenv("PUBSUB_EMULATOR_HOST", "localhost:8085")
+	ctx, _ := context.WithTimeout(context.Background(), 10*time.Second)
+
+	pubsub_cfg := PubSubConfig{GCPProjectID: "Test", GCPPubSubTopic: "TestTopic", GCPSubscriptionName: "TestSub", context: ctx, options: []option.ClientOption{option.WithoutAuthentication()}}
+	setup_client(pubsub_cfg)
+
+	cmp, err := component.NewFromEnv()
+	if err != nil {
+		panic(err)
```

Comment:
Let's just fail the test suite instead of panic so other tests can also fail and report success

`t.Fatalf("component.NewFromEnv failed with err: %v", err)`